XenAdmin/Wizards/CrossPoolMigrateWizard/CrossPoolMigrateDestinationPage.cs

```diff
@@ -148,12 +148,22 @@ protected override DelayLoadingOptionComboBoxItem CreateDelayLoadingOptionComboB
             return new DelayLoadingOptionComboBoxItem(xenItem, filters);
         }
 
-        protected override List<ReasoningFilter> CreateTargetServerFilterList(IEnableableXenObjectComboBoxItem selectedItem, List<VM> vmList)
+        protected override List<ReasoningFilter> CreateTargetServerFilterList(IEnableableXenObjectComboBoxItem selectedItem, List<string> vmOpaqueRefs)
         {
             var filters = new List<ReasoningFilter>();
 
-            if(selectedItem != null)
+            if(selectedItem != null && vmOpaqueRefs != null && selectedVMs != null)
             {
+                List<VM> vmList = new List<VM>();
+                foreach (string _opaqRef in vmOpaqueRefs)
+                {
+                    vmList.AddRange(selectedVMs.FindAll(

```

Comment:
I don't think we need to call `FindAll` here, because there should be only one VM with a given opaque_ref.
So we can use `Find()` and `vmList.Add()`, e.g. 
```
foreach (string opaqRef in vmOpaqueRefs)
     vmList.Add(selectedVMs.Find(vm => vm.opaque_ref == opaqRef));
```

Reply:
Agree.